plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -305,13 +304,13 @@ fu_thunderbolt_device_prepare_firmware(FuDevice *device,
 				       GError **error)
 {
 	FuThunderboltDevice *self = FU_THUNDERBOLT_DEVICE(device);
-	g_autoptr(FuThunderboltFirmwareUpdate) firmware = fu_thunderbolt_firmware_update_new();
-	g_autoptr(FuThunderboltFirmware) firmware_old = fu_thunderbolt_firmware_new();
+	g_autoptr(FuFirmware) firmware = fu_intel_thunderbolt_firmware_new();
```

Comment:
I still fell like this shouldn't be defaulting to Intel FW type.  When it's parsed it looks specifically like an Intel one you subclass it to the Intel one, but it should start out as one that is generic.

Reply:
I put the thing about fu_firmware_new_from_gtypes() into the commit message -- I figured we'd switch from the explicit type to using `fu_firmware_new_from_gtypes` once we had a firmware parser for ASMedia implemented.

Reply:
I guess i'm not convinced we're going to get a parser for ASMedia.  They indicated on the kernel mailing list that they're not disclosing their NVM format (no need to).  So I would rather see it that we match Intel's firmware we know some of the format or "generic" USB4 firmware with no details.  I don't know if we'll be seeing VIA come to the table either with their format.

Reply:
How about the fixup commit? That's what I had in mind.

Reply:
Yes! That's the direction I had in mind too.  The only thing I would change with it is that there should be a type that will match when it's not Intel.  Maybe just "GENERIC_USB4".  Then ASMedia and Via and all will hopefully with no other changes, and if we get parser info we can add that later.

Reply:
> should be a type that will match when it's not Intel

What would it do that isn't covered by "_unknown binary blob_" of FuFirmware?

Reply:
FYI `fu_firmware_check_compatible` does the heavy lifting, and that's a vfunc that can be implemented by any future firmware subclass.

Reply:
> What would it do that isn't covered by "unknown binary blob" of FuFirmware?

Oh perhaps.  I was getting caught up in the checking it's not `NULL`.  If that's not the common path, then yeah this is good.